src/app/api/v1/admin/feature-flags/route.ts

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { NextRequest, NextResponse } from "next/server";
+import { getToken } from "next-auth/jwt";
+import {
+  getAllFeatureFlags,
+  addFeatureFlag,
+  enableFeatureFlagByName,
+  disableFeatureFlagByName,
+  updateAllowList,
+  updateWaitList,
+} from "../../../../../db/tables/featureFlags";
+
+import type { FeatureFlag } from "../../../../../db/tables/featureFlags";
+import { isAdmin } from "../../../utils/auth";
+import appConstants from "../../../../../appConstants";
+
+export async function GET(req: NextRequest) {
+  const token = await getToken({ req });
+  if (typeof token?.email === "string" && isAdmin(token?.email)) {
+    // Signed in
+    try {
+      const flags = await getAllFeatureFlags();
+      return NextResponse.json(flags);
+    } catch (e) {
+      return NextResponse.json({ success: false }, { status: 500 });
+    }
+  } else {
+    // Not Signed in, redirect to home
+    return NextResponse.redirect(appConstants.SERVER_URL, 301);
+  }
+}
+
+export async function POST(req: NextRequest) {
+  const token = await getToken({ req });
+  if (typeof token?.email === "string" && isAdmin(token?.email)) {
+    // Signed in
+    try {
+      const newFlag = (await req.json()) as FeatureFlag;
+      const resp = await addFeatureFlag(newFlag);
+      return NextResponse.json(resp);
+    } catch (e) {
+      console.error(e);
+      return NextResponse.json({ success: false }, { status: 500 });
+    }
+  } else {
+    // Not Signed in, redirect to home
+    return NextResponse.redirect(appConstants.SERVER_URL, 301);
+  }
+}
+
+type FeatureFlagPutRequest = {
+  name: string;
+  isEnabled?: boolean;
+  allowList?: [string];
```

Comment:
Just a note in case it's unintentional: `[string]` is an array with a single element (a `string`), whereas `string[]` (or `Array<string>`) is an array of strings.